docs/CustomExporter.md

```diff
@@ -0,0 +1,72 @@
+Custom Exporter User Guide
+============================
+
+Description
+-----------
+In addition to the existing exporters like the NVIDIA exporter (GPU level), IB exporter (Infiniband level), and node exporter (Node level), Moneo also provides support for a custom exporter, allowing users to expose their own metrics. These custom metrics can include training loss, training accuracy, and more.
+
+Steps
+-----
+1. Create a JSON file to define your custom metrics following a format similar to the examples below:
+
+   - If you want to expose GPU-related metrics for a specific GPU device, use the format `gpu_<gpu_id>_<metric_name>: <metric_value>`. The resulting metric will be named `gpu_<metric_name>` with a label `gpu_id` corresponding to the specified GPU ID.
+
+   - If you want to expose Infiniband (IB) related metrics, use the format `ib_<ib_port>_<metric_name>: <metric_value>`. The resulting metric will be named `ib_<metric_name>` with a label `ib_port` corresponding to the specified IB port.
+
+   - If you simply want to define a metric without any specific label, use the format `<metric_name>: <metric_value>`. This metric will not have any associated labels.
+
+   Here is a template for a custom_metrics_file in JSON format:
+
+   ```json
+   {
+       "gpu_0_metric1": 1.0,
+       "gpu_1_metric1": 1.0,
+       "gpu_2_metric1": 1.0,
+       "gpu_3_metric1": 1.0,
+       "ib_0_metric2": 1.0,
+       "ib_1_metric2": 1.0,
+       "ib_2_metric2": 1.0,
+       "ib_3_metric2": 1.0,
+       "metric3": 1.0,
+       "metric4": 1.0
+   }
+   ```
+
+2. Start Moneo with the following command, specifying the path to your custom metrics file and enabling the custom_exporter on port `8003`:
+
+   ```bash
+   python3 moneo.py -d -c <host_file> --custom_metrics_file_path <custom_metrics_file_path>
```

Comment:
I tried launching with this json file:
{
   "gpu_0_metric1": 1.0,
   "gpu_1_metric1": 1.0,
   "gpu_2_metric1": 1.0
}

the custom exporter did not launch? am I missing something

Reply:
I can give you access to my VM if you would like

Reply:
Thanks for catching this! I have replaced single quotes with double quotes. Not sure this issue can be fixed as I don't have a GPU VM for testing currently.

Reply:
Ack. Let me continue the review and we can figure out how we can test with new fixes.

Reply:
I have tested this on VMs and it works well. Thanks for catching it!